src/dbnode/storage/index/block.go

```diff
@@ -110,10 +110,10 @@ type block struct {
 	blockStart    time.Time
 	blockEnd      time.Time
 	blockSize     time.Duration
+	blockOpts     BlockOptions
 	opts          Options
 	iopts         instrument.Options
 	nsMD          namespace.Metadata
-	docsPool      doc.DocumentArrayPool
```

Comment:
super nit: mind grouping the lazily allocated objects are grouped together in a struct and there's a comment about using the ensure method before using

Reply:
Good call.